packages/components/src/Search.purs

```diff
@@ -68,9 +79,49 @@ search = do
                              Array.concatMap (_.faro) res
               Left _ -> pure unit
             liftEffect $ setSearchWrapper $ AsyncWrapper.Success Nothing
+        startCreateAccount usr@{ cusno, email } = do
+          pw <- randomString 10
+          nowISO <- JSDate.toISOString =<< JSDate.now
+          let legalConsent =
+                { consentId: "legal_acceptance_v1"
+                , screenName: "legalAcceptanceScreen"
+                , dateAccepted: nowISO
+                }
+          setAccountData $ const $ Just $ Tuple cusno $ AsyncWrapper.Editing
+            { cusno: cusno
+            , email: fromMaybe "" email
+            , firstName: ""
+            , lastName: ""
+            , password: pw
+            , consents: [legalConsent]
+            }
+        submitCreateAccount newUser = Aff.launchAff_ do
+          result <- User.createCusnoUser newUser
+          let setError err = setAccountData $ (map <<< map) (const $ AsyncWrapper.Error err)
+          liftEffect $ case result of
+            Left User.RegistrationEmailInUse ->
+              setError "E-postadressen är redan i bruk."
+            Left User.RegistrationCusnoInUse ->
+              setError "Kundnummer är redan i bruk."
+            Left _ ->
+              setError "Något gick fel."
+            Right res -> do
+              let faroCusno x = map _.cusno $ Array.head x.faro
+                  updateWithJanrain :: SearchResult -> SearchResult -> SearchResult
+                  updateWithJanrain x state =
+                    if isNothing state.janrain && faroCusno state == faroCusno x
+                      then x
+                      else state
+              setResults $ (map <<< map) $ updateWithJanrain res
+              setAccountData $ const Nothing
+        createAccountForm = (map <<< map)
```

Comment:
Would you explain what's going on in this function? 😂 It is beautiful though.

Reply:
`accountData`'s type is `Maybe (Tuple Cusno (AsyncWrapper.Progress User.NewCusnoUser)))`, so the `map <<< map` means that it's operating on the `AsyncWrapper.Progress User.NewCusnoUser` part. That matches the fourth argument of `renderEditNewUser`.
`setAccountData` by itself has type `(Maybe (Tuple Cusno (AsyncWrapper.Progress User.NewCusnoUser))) -> Maybe (Tuple Cusno (AsyncWrapper.Progress User.NewCusnoUser)))) -> Effect Unit`. Composing it with `map` peels away those functor layers so that it ends up with just `(User.NewCusnoUser -> User.NewCusnoUser) -> Effect Unit`. That's the third argument of `renderEditNewUser`.